subsys/bluetooth/host/smp.c

```diff
@@ -1813,6 +1811,26 @@ static void smp_reset(struct bt_smp *smp)
 	}
 }
 
+static uint8_t hci_err_get(enum bt_security_err err)
```

Comment:
Why? What is the code path that makes you translate from a security error to an HCI one? This is the Host.

Reply:
It's more clear if you look at the commit: https://github.com/zephyrproject-rtos/zephyr/pull/32120/commits/2352ea33150e10e784622ed34937ec6543304098

Reply:
So it's not an error originating from HCI, rather you're synthesising an HCI error code based on an SMP error which has been first translated into a `bt_security_err`? That sounds like lots of translations to me :)

Reply:
My only concern with this is that when presented with an HCI error somewhere higher up in the API I'd immediately want to take a look at the HCI log, and not finding the corresponding error code there is likely to induce some head scratching :)

Reply:
Completely unrelated to this PR, I wonder if it'd make sense to consider some abstracted error type that could encapsulate errors from multiple different domains: HCI, protocol (like SMP), internal (e.g. based on errno), etc. That way you wouldn't need to translate from one domain to another, rather the application could see in what domain and which layer the error originated from. This might even be a Zephyr-wide discussion since other subsystems may want it as well.

Reply:
I think it might be a good idea if we did something like that, akin to the `k_timeout_t` we introduced in the kernel. But this is long-term.
In the meantime, is there not a way we can store the HCI error code?

Reply:
@jhedberg The bt_security_err is meant to do that, encapsulate the HCI and SMP, the problem is that the security_changed is raising the encrypt_change callback for the l2cap channels, which accepts HCI status code.
In order to update that we would have to update the encrypt change callback signature, which is part of the public API.
`void (*encrypt_change)(struct bt_l2cap_chan *chan, uint8_t hci_status);`

@carlescufi There is no place to store the HCI error code, since we are terminating the security during the pairing procedure, i.e this is happening in SMP.